exporter/awsxrayexporter/go.mod

```diff
@@ -2,8 +2,11 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsxra
 
 go 1.14
 
+replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/common => ../../internal/common
+
 require (
-	github.com/aws/aws-sdk-go v1.34.1
+	github.com/aws/aws-sdk-go v1.34.3
```

Comment:
I think we were avoiding a dependency on aws-sdk-go. If we're only using it for the pointer factories, we don't need it. Otherwise, shouldn't we use the model classes in the SDK itself?

Reply:
1. Not sure I understand what you meant by avoiding `aws-sdk-go`, the x-ray exporter definitely needs the Go SDK, right?
2. We can for sure write our own pointer utilities, but since Amazon's distribution of OpenTelemetry will include the X-Ray exporter, which implies that the Go AWS SDK is always included in the binary, I don't see the need to duplicate a few utilities.
3. There's no model class for X-Ray segment in the SDK. Only [this](https://docs.aws.amazon.com/sdk-for-go/api/service/xray/#PutTraceSegmentsInput).

Reply:
Ah yeah thought we could keep the dependency out of the common package since we're not using the SDK here. 

I think duplicating utilities if they're small is OK - my first impression when reading the code was confusion because we mix `awsxray` and `aws` within the code, which makes it confusing to know which is in this repo vs an external repo because of the similar names. Not a deal breaker, but I don't think the small amount of logic is worth it. A side benefit is if you had a separate package with these small utilities, it would be reasonable to use a period import to reduce the `aws.` type of prefix in the code, but we wouldn't want to do that if it imported in a whole lot of aws-sdk-go under the unprefixed namespace at the same time.

Reply:
Readability is really important. Will update

Reply:
Update: I tried creating a `"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/util"` to put the pointer utilities there so that they can be used by both the `internal` package as well as the receiver, which does not really depends on AWS SDK. Turned out go can't handle that:

```
go: finding module for package github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/util
# github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awsxrayreceiver
package github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awsxrayreceiver (test)
        imports github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/util: module github.com/open-telemetry/opentelemetry-collector-contrib/internal/common@latest found (v0.7.0, replaced by ../../internal/common), but does not contain package github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/util
```

Reverting back to use `aws.String()` and `aws.Bool()`